File: my-shop-selenium/src/test/java/pages/SingInPage.java
package pages;

import data.dto.UserDTO;
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;

public class SingInPage extends BasePage {

    //region Locators
    private static final By emailCreateInput = By.id("email_create");
    private static final By createAccountButton = By.id("SubmitCreate");
    private static final By genderManRadioButton = By.id("id_gender1");
    private static final By genderWomanRadioButton = By.id("id_gender2");
    private static final By firstNameInput = By.id("customer_firstname");
    private static final By lastNameInput = By.id("customer_lastname");
    private static final By emailInput = By.id("email");
    private static final By passwordInput = By.id("passwd");
    private static final By daysSelect = By.id("days");
    private static final By monthsSelect = By.id("months");
    private static final By yearsSelect = By.id("years");
    private static final By registerButton = By.id("submitAccount");
    //endregion

    //region Methods
    public static void fillEmailCreateInput(String email) {
        sendKeys(emailCreateInput, email);
    }

    public static void clickCreateAccountButton() {
        click(createAccountButton);
    }

    public static void clickGenderManRadioButton() {
        click(genderManRadioButton);
    }

    public static void clickGenderWomanRadioButton() {
        click(genderWomanRadioButton);
    }

    public static void fillFirstNameInput(String firstName) {
        sendKeys(firstNameInput, firstName);
    }

    public static void fillLastNameInput(String lastName) {
        sendKeys(lastNameInput, lastName);
    }

    public static void fillEmailInput(String email) {
        sendKeys(emailInput, email);
    }

    public static void fillEmailInput(Keys key) {
        sendKeys(emailInput, key);
    }

    public static void fillPasswordInput(String password) {
        sendKeys(passwordInput, password);
    }

    public static void selectDay(String day) {
        selectValue(daysSelect, day);
    }

    public static void selectMonth(String month) {
        selectValue(monthsSelect, month);
    }

    public static void selectYear(String year) {
        selectValue(yearsSelect, year);
    }

    public static void clickRegisterButton() {
        click(registerButton);
    }
    public static void accessPage() {
        driver.get("http://www.automationpractice.pl/index.php?controller=authentication&back=my-account");
        get("http://www.automationpractice.pl/index.php?controller=authentication&back=my-account");
    }
    //endregion

    //region Fluxos
    public static void registerUser(UserDTO userDTO) {
        fillEmailCreateInput(userDTO.getEmail());
        clickCreateAccountButton();
        if (userDTO.isMen())
            clickGenderManRadioButton();
        else
            clickGenderWomanRadioButton();
        clickGenderManRadioButton();
        fillFirstNameInput(userDTO.getFistName());
        fillLastNameInput(userDTO.getLastName());
        fillPasswordInput(userDTO.getPassword());
        selectDay(userDTO.getDayOfBirth());
        selectMonth(userDTO.getMonthOfBirth());
        selectYear(userDTO.getYearOfBirth());
        clickRegisterButton();
    }
    //endregion
}

File: my-shop-cypress/cypress/e2e/pages/singin-page.ts
import UserDTO from "../data/dto/user-dto";

export default class SingInPage {

    static emailCreateInput = "#email_create";
    static createAccountButton = "#SubmitCreate";
    static creationAlertErrorMessages = "#create_account_error li";
    static formCreateAccount = "#account-creation_form";
    static genderManRadioButton = "#id_gender1";
    static genderWomanRadioButton = "#id_gender2";
    static firstNameInput = "#customer_firstname";
    static lastNameInput = "#customer_lastname";
    static emailInput = "#email";
    static passwordInput = "#passwd";
    static daysSelect = "#days";
    static monthsSelect = "#months";
    static yearsSelect = "#years";
    static registerButton = "#submitAccount";
    static formCreateAlertErrorMessages = "div.alert.alert-danger li";


    static fillEmailCreateInput(email: string) {
        cy.get(this.emailCreateInput).type(email);
    }

    static clickCreateAccountButton = () => {
        cy.get(this.createAccountButton).click();
    }

    static getMeessagesCreationAlertError = () => {
        return cy.get(this.creationAlertErrorMessages)
                    .invoke("text");
    }

    static clickGenderManRadioButton = () => {
        cy.get(this.genderManRadioButton).click();
    }

    static clickGenderWomanRadioButton = () => {
        cy.get(this.genderWomanRadioButton).click();
    }

    static fillFirstNameInput(firstName: string) {
        cy.get(this.firstNameInput).type(firstName);
    }

    static fillLastNameInput(lastName: string) {
        cy.get(this.lastNameInput).type(lastName);
    }

    static fillEmailInput(email: string) {
        cy.get(this.emailInput).type(email);
    }

    static clearEmailInput = () => {
        cy.get(this.emailInput).clear();
    }

    static fillPasswordInput(password: string) {
        cy.get(this.passwordInput).type(password);
    }

    static selectDay(day: string) {
        cy.get(this.daysSelect).select(day);
    }

    static selectMonth(month: string) {
        cy.get(this.monthsSelect).select(month);
    }

    static selectYear(year: string) {
        cy.get(this.yearsSelect).select(year);
    }

    static clickRegisterButton = () => {
        cy.get(this.registerButton).click();
    }

    static getMeessagesFormCreateAlertError = () => {
        return cy.get(this.formCreateAlertErrorMessages);
    }
    
    static accessPage = () => {
        cy.visit(Cypress.config("baseUrl")+"?controller=authentication&back=my-account");
    }

    static waitElement(element: string) {
        cy.get(element).should('be.visible');
    }
    
    
    
    static registerUser(userDTO: UserDTO) {
        this.openForm(userDTO.email);
        
        if (userDTO.isMen)
            this.clickGenderManRadioButton();
        else
            this.clickGenderWomanRadioButton();

        this.clickGenderManRadioButton();
        this.fillFirstNameInput(userDTO.fistName);
        this.fillLastNameInput(userDTO.lastName);
        this.fillPasswordInput(userDTO.password);
        this.selectDay(userDTO.dayOfBirth);
        this.selectMonth(userDTO.monthOfBirth);
        this.selectYear(userDTO.yearOfBirth);
        this.clickRegisterButton();
    }

    static openForm(email: string) {
        this.fillEmailCreateInput(email);
        this.clickCreateAccountButton();
        this.waitElement(this.formCreateAccount);
    }
}